src/Forms/Main.cs

```diff
@@ -1929,13 +1929,12 @@ private void OpenSubtitle(string fileName, Encoding encoding, string videoFileNa
                 if (_subtitle.HistoryItems.Count > 0 || _subtitle.Paragraphs.Count > 0)
                     MakeHistoryForUndo(string.Format(_language.BeforeLoadOf, Path.GetFileName(fileName)));
 
-                bool change = _changeSubtitleToString != _subtitle.GetFastHashCode();
-                if (change)
-                    change = _lastDoNotPrompt != _subtitle.GetFastHashCode();
+                string _subtitleHash = _subtitle.GetFastHashCode();
+                bool hasChanged = (_changeSubtitleToString != _subtitleHash) && (_lastDoNotPrompt != _subtitleHash);
 
                 SubtitleFormat format = _subtitle.LoadSubtitle(fileName, out encoding, encoding);
-                if (!change)
-                    _changeSubtitleToString = _subtitle.GetFastHashCode();
+                if (!hasChanged)
+                    _changeSubtitleToString = _subtitleHash;

```

Comment:
Could you not use the cached hash here.? (it should be from the freshly loaded subtitle)